lib/iris/cube.py

```diff
@@ -2263,8 +2180,18 @@ def new_cell_measure_dims(cm_):
             if ma.count_masked(data) == 0:
                 data = data.filled()
 
+        # XXX: Slicing a single item from a masked array that is masked,
+        #      results in numpy (v1.11.1) *always* returning a MaskedConstant
+        #      with a dtype of float64, regardless of the original masked
+        #      array dtype!
+        if isinstance(data, ma.core.MaskedConstant) and \
+                data.dtype != cube_data.dtype:
```

Comment:
I could. They're both one and the same really. However, for me using `self._data_manager.dtype` is another degree removed from what's happening here ... in that `data` is the result of slicing `cube_data` - I thought that given the context of the problem, it was a wee bit more clearer to reason what's going on and what's expected.

So, I'm erring on the side of the status quo here, unless I'm missing something else ...